src/components/Common/UI/Switch/Switch.tsx

```diff
@@ -0,0 +1,68 @@
+import { Switch as _Switch, type SwitchProps as AriaSwitchProps } from 'react-aria-components'
+import { useFieldIds } from '../hooks/useFieldIds'
+import {
+  HorizontalFieldLayout,
+  type SharedHorizontalFieldLayoutProps,
+} from '../HorizontalFieldLayout'
+import styles from './Switch.module.scss'
+
+export interface SwitchProps
```

Comment:
Could we also update to add an inputRef prop to SwitchProps and hook it up? we end up getting a dom error from hook form if that prop isn't handled

Reply:
Yup will do!